app/Http/Controllers/OrderController.php

```diff
@@ -50,7 +41,33 @@ public function create()
      */
     public function store(Request $request)
     {
-        //
+        $order = new Order();
+        $order->order_code = $request->info_delivery . time();
+
+        $cart = Session::get('cart');
+        $total = 0;
+        foreach ($cart as $key => $product) {
```

Comment:
những phần code này liên quan đến cart thì nên tách ra viết vào cart controller mới đúng chứ nhỉ?

Reply:
đây là hàm tạo order, đoạn này chỉ lấy dữ liệu từ session để tạo order thôi ạ, nên e viết vào order controller

Reply:
đoạn code này là add sản phẩm vào cart, thay đổi số lượng sản phẩm cho cart, thì chẳng liên quan đến order.
Tạo order nó sẽ tương ứng với action new, store của controller order.
add sản phẩm, chỉnh sửa số lượng sản phẩm trong cart thì tương ứng các action new, store, update cart. Và tất nhiên ở đây em có thể dùng session hoặc cookies cho chức năng này

Reply:
Hàm này dùng để tạo order khi người dùng chọn nút thanh toán. Khi đó, em mới thay đổi lại số lượng sản phầm còn lại trong kho, sau đó xóa session chứa cart đi.
Không có thay đổi Cart ở hàm này ạ. Hàm thay đổi sản phầm trong cart em viết ở file CartController.

Reply:
ok em